### PR TITLE
Adds support for ocean currents

### PR DESCRIPTION
**Pull request by arjo129**
Signed-off-by: Arjo Chakravarty <arjo@openrobotics.org>

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

# 🎉 New feature

Closes #<NUMBER>

## Summary
Adds support for ocean currents to the hydrodynamics plugin. It also adds an optional support for namespaces.

The behaviour of the `<namespace>` parameters is described as follows:
```
  ///   <namespace> - This allows the robot to have an individual namespace for
  ///     current. This is useful when you have multiple vehicles in different
  ///     locations and you wish to control the currents of each vehicle
  ///     separately. If no namespace is given then the plugin listens on
  ///     the `/ocean_current` topic for a `Vector3d` message. Otherwise it
  ///     listens on `/model/{namespace name}/ocean_current`.[String, Optional]
```

## Test it

To test run:
```
ign gazebo auv_controls.sdf
```
And apply a _really strong_ current:
```
ign topic -t /ocean_current -m ignition.msgs.Vector3d -p 'x: 10, y:0, z:0'
```
ocean_currents

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: tutorials/underwater_vehicles.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ The behaviour of a moving body through water is different from the behaviour of
 a ground based vehicle. In particular bodies moving underwater experience much
```
Comment (chapulina) [inline tutorials/underwater_vehicles.md:14]: I noticed that this tutorial isn't listed on `tutorials.md.in`, can we add it there?

Also please add `\page underwater_vehicles Underwater vehicles` to the top so the tutorial is correctly generated by doxygen. Thanks!

**Comment by caguero** [comment]
I added very minor tweaks in 20b57f9.

**Review by caguero — CHANGES_REQUESTED** [review]
My main suggestion is to also support the ability to set the current via SDF. This way, we'll cover the most common use case without having to send any additional messages. Remember to document the extra parameter as well.

**Comment by arjo129** [comment]
I've added a `<defaultCurrent>` field.
```
  ///   <defaultCurrent> - A generic current.
  ///      [vector3d m/s, optional, default = [0,0,0]m/s]
```

File: src/systems/hydrodynamics/Hydrodynamics.hh [hunk 2]
```diff
@@ -72,6 +72,8 @@ namespace systems
   ///     separately. If no namespace is given then the plugin listens on
   ///     the `/ocean_current` topic for a `Vector3d` message. Otherwise it
   ///     listens on `/model/{namespace name}/ocean_current`.[String, Optional]
+  ///   <defaultCurrent> - A generic current.
```
Comment (caguero) [inline src/systems/hydrodynamics/Hydrodynamics.hh:75]: Nit: Typically, the SDF parameters are `snake_case`. I recommend to change it to `default_current`.

**Review by caguero — CHANGES_REQUESTED** [review]
Very minor request, other than that looks good.

File: src/systems/hydrodynamics/Hydrodynamics.hh (hunk 2 above)
Comment (arjo129) [reply src/systems/hydrodynamics/Hydrodynamics.hh:75]: Should I also change `<waterDensity>` to `<water_density>`,

Comment (caguero) [reply src/systems/hydrodynamics/Hydrodynamics.hh:75]: You can but leave the original as well (deprecated) and we'll remove in the next version.

Comment (caguero) [reply src/systems/hydrodynamics/Hydrodynamics.hh]: All changes look good to me but my only suggestion is to show a warning or similar if the user uses `waterDensity`. We'll remove support for that parameter in the next release ("G").

**Review by mabelzhang — COMMENTED** [review]
Nits.

**Comment by mabelzhang** [comment]
CI looks passing. I think Windows is disabled (thanks to the awesome new CI documentation in ignitionrobotics/docs/ci.md!), and the failing tests on Mac I've seen before. I think they're flaky.

I haven't tried the PR itself, but we should probably add some integration tests next time (if it's possible to test with some known numbers?), to make sure we don't break stuff with new features.

**Review by caguero — APPROVED** [review]

**Comment by arjo129** [comment]
DCO won't let me merge. @caguero I think you need to sign the last commit.

**Pull request merged**